cfme/services/dashboard/ssui.py

```diff
@@ -0,0 +1,144 @@
+from utils.appliance.implementations.ssui import navigator, SSUINavigateStep, navigate_to, ViaSSUI
+
+from . import Dashboard
+
+
+class DashboardView(SSUIBaseLoggedInPage):
+    title = Text(locator='//li[@class="active"]')
+    dashboard_card = SSUIPrimarycard()
+    aggregate_card = SSUIAggregatecard()
+
+    def in_dashboard(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["", "Dashboard"])
+
+    @property
+    def is_displayed(self):
+        return self.in_dashboard
+
+
+class MyServiceForm(SSUIBaseLoggedInPage):
+
+    service = SSUIlist(list_name='serviceList')
+
+
+class MyServicesView(MyServiceForm):
+    title = Text(locator='//li[@class="active"]')
+
+    def in_myservices(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["", "My Services"])
+
+    @property
+    def is_displayed(self):
+        return self.in_myservices and self.title.text == "My Services"
+
+
+@Dashboard.total_service.external_implementation_for(ViaSSUI)
+def total_service(self):
+    view = navigate_to(self, 'TotalServices')
+    view.flash.assert_no_error()
+    view = self.create_view(MyServicesView)
+    assert view.is_displayed
+
+
+@Dashboard.total_request.external_implementation_for(ViaSSUI)
+def total_request(self):
+    """ Returns the total number of requests displayed on dashboard."""
+
+    view = navigate_to(self, 'DashboardAll')
+    total_request = view.dashboard_card.get_count('Total Requests')
+    return total_request
+
+
+@Dashboard.retiring_soon.external_implementation_for(ViaSSUI)
+def retiring_soon(self):
+    view = navigate_to(self, 'DashboardAll')
+    view.aggregate_card.get_count('Retiring Soon')
+    view = navigate_to(self, 'RetiringSoon')
+    view.flash.assert_no_error()
+    view = self.create_view(MyServicesView)
+    assert view.is_displayed
+
+
+@Dashboard.current_services.external_implementation_for(ViaSSUI)
+def current_services(self):
+    view = navigate_to(self, 'DashboardAll')
+    view.aggregate_card.get_count('Current Services')
+    view = navigate_to(self, 'CurrentServices')
+    view.flash.assert_no_error()
+    view = self.create_view(MyServicesView)
+    assert view.is_displayed
+
+
+@Dashboard.retired_services.external_implementation_for(ViaSSUI)
+def retired_services(self):
+    view = navigate_to(self, 'DashboardAll')
+    view.aggregate_card.get_count('Retired Services')
+    view = navigate_to(self, 'RetiredServices')
+    view.flash.assert_no_error()
+    view = self.create_view(MyServicesView)
+    assert view.is_displayed
+
+
+@Dashboard.monthly_charges.external_implementation_for(ViaSSUI)
+def monthly_charges(self):
+    view = navigate_to(self, 'DashboardAll')
+    chargeback = view.aggregate_card.get_count('Monthly Charges - This Month To Date')
+    return chargeback
+
+
+@navigator.register(Dashboard, 'DashboardAll')
+class DashboardAll(SSUINavigateStep):
```

Comment:
**REQUIRED**: This navigation destination in particular doesn't work. The login page matches this and so if navigation is interrupted it can't then get past the login page as it thinks it's already on DashboardAll

Reply:
Added more checks for dashboard view